test/plugins.spec.ts

```diff
@@ -0,0 +1,153 @@
+import {
+  ICreatePlugin,
+  IGetters,
+  IModule,
+  IMutations,
+  IPlugin,
+  IState,
+  LocalStoragePlugin,
+  ILocalStoragePluginState,
+  flatten,
+  createModule,
+} from '../lib'
+
+interface IBaseTestState extends IState {
+  id: number
+}
+
+interface IOldTestState extends IBaseTestState {
+  name: string
+}
+
+interface INewTestState extends IBaseTestState {
+  age: number
+}
+
+interface ITestGetters extends IGetters {
+  isDefault: () => boolean
+}
+
+interface ITestMutations extends IMutations {
+  updateId: (id: number) => void
+}
+
+const getOldDefaultState = (): IOldTestState => ({
+  id: 0,
+  name: '',
+})
+
+const getNewDefaultState = (): INewTestState => ({
+  id: 0,
+  age: 5,
+})
+
+interface ITestModuleOptions<S extends IBaseTestState> {
+  version: number
+  stateInit: () => S
+  plugins: (ICreatePlugin<S> | IPlugin<S>)[]
+}
+
+const makeTestModule = <S extends IBaseTestState>({
+  version,
+  stateInit,
+  plugins,
+}: ITestModuleOptions<S>): IModule<S, ITestGetters, ITestMutations> =>
+  createModule<S, ITestGetters, ITestMutations>({
+    name: 'test',
+    version,
+    stateInit,
+    getters: (state) => ({
+      isDefault: () => state.id === 0 && state.name === '',
+    }),
+    mutations: (state) => ({
+      updateId: (id: number) => {
+        state.id = id
+      },
+    }),
+    plugins,
+  })
+
+describe('vue-store plugins', () => {
+  beforeEach(() => {
+    localStorage.clear()
+    jest.clearAllMocks()
+  })
+
+  describe('local stroage plugin', () => {
+    it('modify state and save with built-in local storage plugin', async () => {
+      const module = flatten(
+        makeTestModule<IOldTestState>({
+          version: 1,
+          stateInit: getOldDefaultState,
+          plugins: [LocalStoragePlugin],
+        }),
+      )
+      expect(module.id.value).toEqual(0)
+
+      // Verify values in localStorage before anything changes
+      const stringifiedStateBefore = localStorage.getItem('test')
+      expect(stringifiedStateBefore).toBeNull()
+
+      // Update id
+      const newId = 1
+      module.updateId(newId)
+      // a workaround to wait until the Vue watch-effect is done
+      await Promise.resolve()
```

Comment:
I noticed that an `await` statement can do the same thing as what we did before:

```ts
await new Promise((resolve) => {
  const timeoutId = setTimeout(() => {
    clearTimeout(timeoutId)
    return resolve(true)
  }, 1)
})
```

`await null` will also work (check [MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/await#control_flow_effects_of_await) for more info, just for reference), but that'll look strange. I guess the watch-effect is probably something related to microtask.

Reply:
Good idea!